### DISCUSSION TITLE
--with-pattern Inserting line breaks in output

### DISCUSSION THREAD

**Discussion by davidSchlosser** [quodlibet/quodlibet#4244] [Q&A] [answered]
Quodlibet's new `--print-query --with-pattern` argument is a great feature for my needs - many thanks to the developers @nemethf !

I can produce output I might be able to use for my needs. What keyboard character would I put in the pattern string to get the line breaks? (\n, \r don't work, I used cut & paste from a 'soft-hyphen' character in a tag pattern I saw used in the mqtt plug-in but I don't know how to type it from a keyboard.)
```
quodlibet/quodlibet.py --print-query='' --with-pattern='artist=<artist>               
genre=<genre>
composer=<composer>
mood=<mood>
style=<style>'  | sort | uniq
```
Sample output:
```
artist=Amaan Ali Bangash               
artist=Amjad Ali Khan               
artist=Art Blakey's Jazz Messengers               
artist=Ayaan Ali Bangash               
artist=Béla Fleck & The Flecktones               
artist=David Oistrakh, Mstislav Rostropovich, Sviatoslav Richter, Herbert von Karajan, Berlin Philarmonic Orchestra               
artist=Mstislav Rostropovich, David Oistrakh, George Szell, Cleveland Orchestra               
composer=
composer=Amjad Ali Khan
composer=Béla Fleck
composer=Johannes Brahms
composer=Ludwig van Beethoven
genre=Classical
genre=Jazz
genre=World
mood=
style=
style=Concerto
```

**Comment by nemethf** [comment, marked as answer]
I don't understand your question.  This does work:
```
QL  --print-playing '<artist>
<title>'
```
But if you want to post-process the result, you might be better off with `--with-pattern <~json>`.

Also your example can be rewritten as:
```
for tag in artist composer genre mood style; do
  QL --print-query '' --with-patern "$tag=<$tag>" |sort|uniq
done
```
which executes more queries, but is more readable, IMHO.

**Comment by davidSchlosser** [comment]
Thanks for your reply. I'm not sure what I was doing wrong yesterday, but, as you rightly say, line breaks in the pattern do work.

I hadn't encountered the `<~json>` tag before, might be very useful also to save the additional process step I have now. **Is there a way to control which tags are included in the json it generates?**

The for loop you recommend is quite readable but would require too many passes over the massive library I'm using.

**Reply by nemethf** [reply]
>  **Is there a way to control which tags are included in the json it generates?**

No, see https://github.com/quodlibet/quodlibet/issues/4191#issuecomment-1296185400

**Reply by davidSchlosser** [reply]
Ok. I can manage that. Thanks again.
